hs_geo_raster_resource/migrations/custom_migration_for_raster_meta_update_20160512.py

```diff
@@ -0,0 +1,139 @@
+
+                    try:
+                        with open(os.devnull, 'w') as fp:
+                            subprocess.Popen(['gdal_translate', '-of', 'VRT', tif_file_path, vrt_file_path], stdout=fp, stderr=fp).wait()   # remember to add .wait()
+
+                        # delete vrt res file
+                        for f in res.files.all():
+                            if 'vrt' == f.resource_file.name[-3:]:
+                                f.resource_file.delete()
+                                f.delete()
+
+                        # add new vrt file to resource
+                        new_file = UploadedFile(file=open(vrt_file_path, 'r'), name=os.path.basename(vrt_file_path))
+                        hydroshare.add_resource_files(res.short_id, new_file)
+
+                        # update the bag
+                        bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                        if istorage.exists(bag_name):
+                            # delete the resource bag as the old bag is not valid
+                            istorage.delete(bag_name)
+                        resource_modified(res, res.creator)
+
+                    except:
+                        print 'fail to update VRT'
+
+                # metadata extraction from temp dir
+                ori_dir = os.getcwd()
+                os.chdir(temp_dir)
+                vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == os.path.splitext(f)[1]].pop()
+                res_md_dict = raster_meta_extract.get_raster_meta_dict(vrt_file_path)
+                os.chdir(ori_dir)
+                shutil.rmtree(temp_dir)
+
+                # update band metadata of raster resources
+                is_modified = False
+                if res_md_dict['band_info']:
+                    for i, band_meta in res_md_dict['band_info'].items():
+                        band_obj = res.metadata.bandInformation.filter(name='Band_{}'.format(str(i))).first()
+                        if band_obj:
+                            res.metadata.update_element('bandInformation',
+                                                        band_obj.id,
+                                                        maximumValue=band_meta['maximumValue'],
+                                                        minimumValue=band_meta['minimumValue'],
+                                                        noDataValue=band_meta['noDataValue'],
+                                                        )
+                            is_modified = True
+
+                # resource modify update
+                if is_modified:
+                    bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                    if istorage.exists(bag_name):
+                        # delete the resource bag as the old bag is not valid
+                        istorage.delete(bag_name)
+                    resource_modified(res, res.creator)
```

Comment:
@gantian127 Again, bag deletion is not needed - setting resource_modified() is sufficient.